tests/sharedv4/sharedv4_test.go

```diff
@@ -0,0 +1,269 @@
+
+		Expect(err).NotTo(HaveOccurred(),
+			fmt.Sprintf("Failed to rescan specs from %s for storage provider %s. Error: [%v]",
+				Inst().SpecDir, provider, err))
+
+		Step("schedule application with multiple sharedv4-svc volumes attached", func() {
+			logrus.Infof("Number of Volumes to be mounted: %v", frequency)
+
+			taskName := "sharedv4-svc-multivol"
+
+			logrus.Infof("Task name %s\n", taskName)
+
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+				newContexts := ScheduleApplications(taskName)
+				contexts = append(contexts, newContexts...)
+			}
+
+			for _, ctx := range contexts {
+				ctx.ReadinessTimeout = timeout
+				ctx.SkipVolumeValidation = false
+				ValidateContext(ctx)
+
+			}
+		})
+
+		Step("get nodes where app is running and restart volume driver", func() {
+			for _, ctx := range contexts {
+				appVolumes, err := Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				for _, appVolume := range appVolumes {
+					attachedNode, err := Inst().V.GetNodeForVolume(appVolume, defaultCommandTimeout, defaultCommandRetry)
+					Expect(err).NotTo(HaveOccurred())
+					Step(
+						fmt.Sprintf("stop volume driver %s on app %s's node: %s",
+							Inst().V.String(), ctx.App.Key, attachedNode.Name),
+						func() {
+							StopVolDriverAndWait([]node.Node{*attachedNode})
+						})
+
+					Step(
+						fmt.Sprintf("starting volume %s driver on app %s's node %s",
+							Inst().V.String(), ctx.App.Key, attachedNode.Name),
+						func() {
+							StartVolDriverAndWait([]node.Node{*attachedNode})
+						})
+
+					Step("Giving few seconds for volume driver to stabilize", func() {
+						time.Sleep(20 * time.Second)
+					})
+
+					Step(fmt.Sprintf("validate app %s", attachedNode.Name), func() {
+						ValidateContext(ctx)
+					})
+				}
+			}
+		})
+
+	})
+
+})
+
+// This test performs sharedv4 nfs server pod termination failover use case
+var _ = Describe("{NfsServerKill}", func() {
+	var contexts []*scheduler.Context
+
+	It("has to validate that the new pods started successfully after nfs server node is terminated", func() {
+		contexts = make([]*scheduler.Context, 0)
+		nodesToKill := []node.Node{}
+		var err error
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("nfskill-%d", i))...)
+		}
+
+		ValidateApplications(contexts)
+		for _, ctx := range contexts {
+			var appVolumes []*volume.Volume
+			Step(fmt.Sprintf("get volumes for %s app", ctx.App.Key), func() {
+				appVolumes, err = Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(appVolumes).NotTo(BeEmpty())
+			})
+			for _, v := range appVolumes {
+
+				Step("get attached node and stop the instance", func() {
+					currNodes := node.GetStorageDriverNodes()
+					countOfCurrNodes := len(currNodes)
+
+					attachedNode, err := Inst().V.GetNodeForVolume(v, defaultCommandTimeout, defaultCommandRetry)
+					nodesToKill = append(nodesToKill, *attachedNode)
```

Comment:
Not sure what is the intention here. Why not just pass only the "attachedNode" to KillANodeAndValidate() function? Why do we append the node each time to the list and then pick a random node to kill?

Reply:
fixed